src/segment-loader.js

```diff
@@ -1212,15 +1215,14 @@ export default class SegmentLoader extends videojs.EventTarget {
     });
   }
 
-  processMetadataQueue_(simpleSegment, type) {
-    if (type !== 'id3' && type !== 'caption') {
-      return;
-    }
+  processMetadataQueue_() {
+    const id3Queue = this.metadataQueue_.id3;
```

Comment:
Any specific reason for creating the variables then executing the functions instead of just doing:
```
this.metadataQueue_.id3.forEach...
this.metadataQueue_.id3 = [] // or .length = 0 if we want to keep the reference
```

Should make for fewer variables and a shorter function.

Reply:
Oh, I think this is just copypasta-ish decision making

Reply:
Good call, but I think if we run into a case like that we probably are doing something wrong elsewhere.

Reply:
Agreed, for this to happen we would have to call this method before we have actually appended some data